lang/src/main/antlr4/org/kaazing/k3po/lang/parser/v2/Robot.g4

```diff
@@ -41,7 +41,11 @@ acceptableNode
     ;
 
 connectNode
-    : k=ConnectKeyword (AwaitKeyword barrier=Name ConnectKeyword)? connectURI=location (OptionKeyword TransportKeyword value=location)? streamableNode+
+    : k=ConnectKeyword (AwaitKeyword barrier=Name ConnectKeyword)? connectURI=location
+                       (OptionKeyword TransportKeyword value=location)?
+                       (OptionKeyword SizeKeyword DecimalLiteral)?
+                       (OptionKeyword ModeKeyword ModeValue)?
+        streamableNode+
     ;
```

Comment:
In general LGTM, but I'm not super familiar with random access file API.  One concern is that we are adding grammar nodes that might not make sense.  e.g. `connect tcp://localhost option mode rw` doesn't make sense for tcp right?  

This is sort of an outstanding issue with the grammar anyways (you can have http keywords in tcp and it still compiles, though you get other exceptions when k3po tries to setup the pipeline).  So bonus points if you can have the option only available in the case it is a file uri.